### PR TITLE
MG-1967 - Handle errors in Things gRPC server and client

### PR DESCRIPTION
**Pull request by arvindh123**
# What type of PR is this?

## What does this do?
Handling of things gRPC errors in both server and client 

## Which issue(s) does this PR fix/relate to?

- Related Issue absmach/magistrala#1967 

## Have you included tests for your changes?

Test partially, Websocket and HTTP doesn

## Did you document any new/modified feature?

No, Not related

### CODE REVIEW

File: things/service.go [hunk 1]
```diff
@@ -578,10 +578,10 @@ func (svc service) Identify(ctx context.Context, key string) (string, error) {
 
 	client, err := svc.clients.RetrieveBySecret(ctx, key)
 	if err != nil {
-		return "", errors.Wrap(repoerr.ErrNotFound, err)
+		return "", errors.Wrap(svcerr.ErrAuthorization, err)
```
Comment (rodneyosodo) [inline things/service.go:581]: Shouldn't this remain to NotFound ?

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: things/service.go (hunk 1 above)
Comment (arvindh123) [reply things/service.go:581]: From the point of Authorization, If the secret is not found then it is invalid secret.

File: things/api/grpc/client.go [hunk 2]
```diff
@@ -69,3 +73,30 @@ func encodeAuthorizeRequest(_ context.Context, grpcReq interface{}) (interface{}
...
+			return errors.Wrap(errors.ErrMalformedEntity, errors.New(st.Message()))
+		case codes.AlreadyExists:
+			return errors.Wrap(errors.ErrConflict, errors.New(st.Message()))
+		case codes.Unauthenticated:
+			return errors.Wrap(errors.ErrAuthentication, errors.New(st.Message()))
+		case codes.OK:
+			if msg := st.Message(); msg != "" {
+				return errors.Wrap(errors.ErrUnidentified, errors.New(msg))
+			}
+			return nil
```
Comment (dborovcanin) [inline things/api/grpc/client.go:92]: Is it likely that the OK code will happen with the nil error? I suspect this case should be either first or last. Also, order cases by the probability (according to your experience) from most frequent to least.

**Review by dborovcanin — CHANGES_REQUESTED** [review]

File: things/api/grpc/client.go (hunk 2 above)
Comment (arvindh123) [reply things/api/grpc/client.go]: If there is no error, we return nill error in MG codes
But gRPC have option to send not error in error status
So only just handled it.

Moved to last case

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**